gallery/tutorials/pipeline_demo.py

```diff
@@ -125,60 +122,78 @@
+# the images for denoising via Class Averaging.
+
+# %%
+# Downsampling
+# ------------
+# We downsample the images to remove high frequency noise and improve the
+# efficiency of subsequent pipeline stages. Metadata such as pixel size is
+# scaled accordingly to correspond correctly with the image resolution.
+
+src = src.downsample(res)
+src.images[:10].show()
 
 # %%
 # CTF Correction
 # --------------
 # We apply ``phase_flip()`` to correct for CTF effects.
 
 src = src.phase_flip()
+src.images[:10].show()
+
+# %%
+# Normalize Background
+# --------------------
+# We apply ``normalize_background()`` to prepare the image class averaging.
+
+src = src.normalize_background()
+src.images[:10].show()
 
 # %%
-# Cache
-# -----
+# Caching
+# -------
 # We apply ``cache`` to store the results of the ``ImageSource``
 # pipeline up to this point.  This is optional, but can provide
 # benefit when used intently on machines with adequate memory.
-
+# Since the remaining preprocessing steps, whitening and contrast
+# inversion, as well as class averaging require passing over the
+# full set of images, caching at this point saves compute time.
 src = src.cache()
-src.images[0:10].show()
+
+# %%
+# Noise Whitening
+# ---------------
+# We apply ``whiten()`` to estimate and whiten the noise.
+
+src = src.whiten()
+src.images[:10].show()
+
+# %%
+# Contrast Inversion
+# ------------------
+# We apply ``invert_contrast()`` to ensure a positive valued signal.
+
+src = src.invert_contrast()
 
 # %%
 # Class Averaging
 # ---------------
-# We use ``RIRClass2D`` object to classify the images via the
-# rotationally invariant representation (RIR) algorithm. Class
-# selection is customizable. The classification module also includes a
-# set of protocols for selecting a set of images to be used after
-# classification.  Here we're using the simplest
-# ``DebugClassAvgSource`` which internally uses the
-# ``TopClassSelector`` to select the first ``n_classes`` images from
-# the source.  In practice, the selection is done by sorting class
-# averages based on some configurable notion of quality (contrast,
-# neighbor distance etc).
-
-from aspire.classification import RIRClass2D
-
-# set parameters
-n_classes = 200
-n_nbor = 6
-
-# We will customize our class averaging source. Note that the
-# ``fspca_components`` and ``bispectrum_components`` were selected for
-# this small tutorial.
-rir = RIRClass2D(
-    src,
-    fspca_components=40,
-    bispectrum_components=30,
-    n_nbor=n_nbor,
-)
+# For this tutorial we use the ``DebugClassAvgSource`` to generate an ``ImageSource``
+# of class averages. Internally, ``DebugClassAvgSource`` uses the ``RIRClass2D``
+# object to classify the source images via the rotationally invariant representation
+# (RIR) algorithm and the ``TopClassSelector`` object to select the first ``n_classes``
+# images from the source. In practice, class selection is commonly done by sorting class
```

Comment:
maybe more like 

to select the first n_classes images in the original order from the source

Then add something to the effect of "which changes the ordering of the classes returned by ClassAverageSource" around the end.

Reply:
Sure. Done. Thanks.